StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp

```diff
@@ -358,66 +361,152 @@ void EnemyFactory::Update(double dt)
-				}
+				eligblePoints.push_back(i);
 			}
+		}
 
-			unsigned int point = m_RandGen.Rand(0, eligblePoints.size());
-			int spawnNumber = m_RandGen.Rand(1, 100);
-			int spawnChance = 0;
-			bool spawnDefault = true;
-			for (auto enemy : m_EnemyComps)
+		unsigned int point = m_RandGen.Rand(0, eligblePoints.size());
+		int spawnNumber = m_RandGen.Rand(1, 100);
+		int spawnChance = 0;
+		bool spawnDefault = true;
+		for (auto enemy : m_EnemyComps)
+		{
+			if (enemy.second != nullptr)
 			{
-				if (enemy.second != nullptr)
+				spawnChance += enemy.second->spawnChance;
+				if (spawnNumber <= spawnChance)
 				{
-					spawnChance += enemy.second->spawnChance;
-					if (spawnNumber <= spawnChance)
-					{
-						SpawnEnemy(enemy.first, eligblePoints[point]);
-						spawnDefault = false;
-						break;
-					}
+					SpawnEnemy(enemy.first, eligblePoints[point]);
+					spawnDefault = false;
+					break;
 				}
 			}
-			if (spawnDefault)
+		}
+		if (spawnDefault)
+		{
+			SpawnEnemy("enemyZombie", eligblePoints[point]);
+		}
+
+		m_EnemiesToSpawn--;
+	}
+}
+void EnemyFactory::killRound(double dt)
+{
+	//m_EnemiesToSpawn hold how many enemies are left to spawn
+		//This exist to stagger the spawning of enemies by one frame to spread the load over multiple frames
+	if (m_EnemiesToSpawn == 0 && m_EnemySlotsLeft > 0)
+	{
+		m_SpawnTimer += dt / m_Enemies.size();
+		if (m_SpawnCooldown <= m_SpawnTimer)
+		{
+			m_SpawnTimer = 0.0;
+			m_EnemiesToSpawn = 3 + m_Level / 3.0f;
+
+			if (m_EnemiesToSpawn > m_EnemySlotsLeft)
+			{
+				m_EnemiesToSpawn = m_EnemySlotsLeft;
+			}
+		}
+	}
+	else if (m_EnemySlotsLeft > 0)
+	{
+		std::vector<int> eligblePoints;
+		float3 playerPos = m_pScene->GetEntity("player")->GetComponent<component::TransformComponent>()->GetTransform()->GetRenderPositionFloat3();
+		for (int i = 0; i < m_SpawnPoints.size(); i++)
+		{
+			float distToPlayer = (m_SpawnPoints[i] - playerPos).length();
+			if (distToPlayer > m_MinimumDistanceToPlayer)
 			{
-				SpawnEnemy("enemyZombie", eligblePoints[point]);
+				eligblePoints.push_back(i);
 			}
+		}
 
-			m_EnemiesToSpawn--;
-			m_EnemySlotsLeft--;
+		unsigned int point = m_RandGen.Rand(0, eligblePoints.size());
+		int spawnNumber = m_RandGen.Rand(1, 100);
+		int spawnChance = 0;
+		bool spawnDefault = true;
+		for (auto enemy : m_EnemyComps)
+		{
+			if (enemy.second != nullptr)
+			{
+				spawnChance += enemy.second->spawnChance;
+				if (spawnNumber <= spawnChance)
+				{
+					SpawnEnemy(enemy.first, eligblePoints[point]);
+					spawnDefault = false;
+					break;
+				}
+			}
+		}
+		if (spawnDefault)
+		{
+			SpawnEnemy("enemyZombie", eligblePoints[point]);
 		}
+
+		m_EnemiesToSpawn--;
+		m_EnemySlotsLeft--;
 	}
 }
 
 void EnemyFactory::enemyDeath(Death* evnt)
 {
-	if (strcmp(evnt->ent->GetName().substr(0, 5).c_str(), "enemy") == 0)
+	//We don't care about kills on time rounds
+	if (strcmp(evnt->ent->GetName().substr(0, 5).c_str(), "enemy") == 0 && !m_TimeRound)
```

Comment:
Better to use string.find()

Reply:
i don't know if it is better and current solution works